### PR TITLE
Post Editor: move Redux action dispatches on post load/save to lib/posts/actions

### PR DESCRIPTION
**Pull request by jsnajdr**
When the Post Editor loads a post to edit (RECEIVE_POST_TO_EDIT) or receives a response to save request (RECEIVE_POST_BEING_EDITED) then we dispatch a few Redux actions to keep Redux in sync with Flux.

This patch moves the dispatches to the action "thunks" in `lib/posts/actions` and uses the Redux bridge to perform them. Previously they were in `PostEditor` lifecycle methods, which makes the code hard to follow and reason about. Further Redux migration will also be much easier now.

**How to test:**
There should be no changes in post editor behavior. Test the following scenarios:
1. Start editing a new draft, enter some content. Verify that it autosaves correctly, that the URL changes to one that includes the assigned post ID and that the dirty flag gets correctly updated (watch the "Save", "Saving..." and "Saved" labels in masterbar) as you modify the post.
2. Edit and existing post -- do the modifications get saved correctly?
3. Publish a post
4. Edit a post copy

### CODE REVIEW

**Comment by jsnajdr** [comment]
The full e2e suite failed with an error in domains management that's very unlikely to be related. Rerunning.

**Comment by jsnajdr** [comment]
After rebasing to latest master, the full e2e suite finally managed to run with all tests passing.

**Review by blowery — APPROVED** [review]
Sweet. Yay for bridges.
